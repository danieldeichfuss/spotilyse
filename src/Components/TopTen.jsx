import React from "react";
import ReactCSSTransitionGroup from "react-addons-css-transition-group";

export default function TopTen({ tracks, artists }) {
  const tracksToRender = tracks.map((track, i) => (
    <li className="TopTen__item" key={i.toString()}>
      <span>{track.name}</span>
    </li>
  ));
  const artistsToRender = artists.map((artist, i) => (
    <li className="TopTen__item" key={i.toString()}>
      <span>{artist.name}</span>
    </li>
  ));
  return (
    <div className="TopTen">
      <h2 className="TopTen__title">Your Top 10</h2>
      <h3 className="TopTen__sub-title">Top 10 Tracks</h3>
      <ReactCSSTransitionGroup
        transitionName="AppearTransition"
        transitionAppear={true}
        transitionAppearTimeout={1000}
        transitionEnter={false}
        transitionLeave={false}
      >
        <ol className="TopTen__list">{tracksToRender}</ol>
      </ReactCSSTransitionGroup>
      <h3 className="TopTen__sub-title">Top 10 Artists</h3>
      <ReactCSSTransitionGroup
        transitionName="AppearTransition"
        transitionAppear={true}
        transitionAppearTimeout={1000}
        transitionEnter={false}
        transitionLeave={false}
      >
        <ol className="TopTen__list">{artistsToRender}</ol>
      </ReactCSSTransitionGroup>
    </div>
  );
}
