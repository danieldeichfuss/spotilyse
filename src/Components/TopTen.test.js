import React from "react";
import ReactDOM from "react-dom";
import TopTen from "./TopTen";
import * as Enzyme from "enzyme";
import ReactSixteenAdapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new ReactSixteenAdapter() });

const tracks = [];
const artists = [];

const wrapper = Enzyme.shallow(<TopTen tracks={tracks} artists={artists} />);

it("should render the component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
