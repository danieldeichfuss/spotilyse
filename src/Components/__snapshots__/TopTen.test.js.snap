// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopTen
    artists={Array []}
    tracks={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="TopTen__title"
>
          Your Top 10
</h2>,
        <h3
          className="TopTen__sub-title"
>
          Top 10 Tracks
</h3>,
        <List
          items={Array []}
/>,
        <h3
          className="TopTen__sub-title"
>
          Top 10 Artists
</h3>,
        <List
          items={Array []}
/>,
      ],
      "className": "TopTen",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Your Top 10",
          "className": "TopTen__title",
        },
        "ref": null,
        "rendered": "Your Top 10",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Top 10 Tracks",
          "className": "TopTen__sub-title",
        },
        "ref": null,
        "rendered": "Top 10 Tracks",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "items": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Top 10 Artists",
          "className": "TopTen__sub-title",
        },
        "ref": null,
        "rendered": "Top 10 Artists",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "items": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="TopTen__title"
>
            Your Top 10
</h2>,
          <h3
            className="TopTen__sub-title"
>
            Top 10 Tracks
</h3>,
          <List
            items={Array []}
/>,
          <h3
            className="TopTen__sub-title"
>
            Top 10 Artists
</h3>,
          <List
            items={Array []}
/>,
        ],
        "className": "TopTen",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Your Top 10",
            "className": "TopTen__title",
          },
          "ref": null,
          "rendered": "Your Top 10",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Top 10 Tracks",
            "className": "TopTen__sub-title",
          },
          "ref": null,
          "rendered": "Top 10 Tracks",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "items": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Top 10 Artists",
            "className": "TopTen__sub-title",
          },
          "ref": null,
          "rendered": "Top 10 Artists",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "items": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
